src/commands/multi_block/monitor.rs

```diff
@@ -148,11 +228,119 @@ where
     }
 }
 
+/// Dedicated task to clear past submissions
+async fn run_clearing_task(
+    client: Client,
+    signer: Signer,
+    mut clear_receiver: mpsc::Receiver<RoundToClear>,
+    submitted_rounds: RoundSubmission,
+    notify: Arc<Notify>,
+) -> Result<(), Error> {
+    // Initial queue size update
+    prometheus::set_clearing_round_queue_size(clear_receiver.len());
+
+    loop {
+        // Process any rounds in the queue
+        let mut processed = false;
+
+        while let Ok(round_to_clear) = clear_receiver.try_recv() {
```

Comment:
this will essentially poll the receiver, let's the async API to avoid waking up the task that often when there is nothing to consume

```suggestion
        while let Ok(round_to_clear) = clear_receiver.recv().await {
```

Reply:
Done a variant of this in https://github.com/paritytech/polkadot-staking-miner/pull/1029/commits/4a7899a23782f711372152ac6c756c80bcb3e70c